index.js

```diff
@@ -23,7 +11,21 @@ async function subscribe(topic_name, subscription_name, subscriber, options) {
     },
   };
 
-  const subscription = pubsub.subscription(subscription_name, options || defaultOptions);
+  const topic = pubsub.topic(topic_name);
+  const subscription = topic.subscription(subscription_name, options || defaultOptions);
+
+  try {
+    const [exists] = await subscription.exists();
+    if (!exists) {
+      logger.info(`Creating subscription ${subscription_name} on topic ${topic_name}`);
+      await subscription.create();
+      logger.info(`Subscription ${subscription_name} on topic ${topic_name} created successfully`);
+    }
+  } catch(err) {
+    logger.error(`Error checking subscription exists or creating subscription`, err);
+    message.nack();
+    return;
+  }
```

Comment:
1) The message is undefined here. if the topic is not there nor created there is no message flow.
2) If the message does not exist nor can be created we should let it fail and not return as if everything is fine.
3) We don't need to try...catch these lines.